react/src/views/Box/BoxView.test.tsx

```diff
@@ -0,0 +1,416 @@
+                                ],
+                            },
+                        },
+                        size: {
+                            id: "52",
+                            label: "Mixed",
+                        },
+                        state: "InStock",
+                        tags: [],
+                    },
+                },
+            },
+        },
+    ];
+
+    const waitTillLoadingIsDone = async () => {
+        await waitFor(() => {
+            const loadingInfo = screen.queryByText("Loading...");
+            expect(loadingInfo).toBeNull();
+        });
+    };
+
+    beforeEach(() => {
+        render(<BTBox />, {
+            routePath: "/bases/:baseId/boxes/:labelIdentifier",
+            initialUrl: "/bases/2/boxes/189123",
+            mocks,
+        });
+    });
+    // Test case 3.1.1.0
+    it("3.1.1.0 - renders with an initial 'Loading...'", async () => {
+        await waitFor(waitTillLoadingIsDone);
+        const loadingInfo = screen.getByTestId("loading-indicator");
+        expect(loadingInfo).toBeInTheDocument();
+    });
+    // Test case 3.1.1.1
+    it("3.1.1.1 - renders Heading with valid box identifier", async () => {
+        await waitFor(waitTillLoadingIsDone);
+        const boxHeader = screen.getByTestId("box-header");
+        expect(boxHeader).toHaveTextContent("Box 189123");
+    });
+    // Test case 3.1.1.2
+    it("3.1.1.2 - renders sub heading with valid state", async () => {
+        await waitFor(waitTillLoadingIsDone);
+        const boxSubheading = screen.getByTestId("box-subheader");
+        expect(boxSubheading).toHaveTextContent("State: Lost");
+    });
+    // Test case 3.1.1.2.1
+    it("3.1.1.2.1 - change box state color respectfully", async () => {
+        await waitFor(waitTillLoadingIsDone);
+        let color;
+        const currentState = mocks[0].result.data.box?.state;
+        if (currentState === "Lost" || currentState === "Scrap") {
+            color = "#EB404A";
+        } else {
+            color = "#0CA789";
+        }
+        expect(screen.getByTestId("box-state")).toHaveStyle(`color: ${color}`);
+    });
+    // Test case 3.1.1.3
+    it("3.1.1.3 - click on + and - to increase or decrease number of items", async () => {
+        await waitFor(waitTillLoadingIsDone);
+        let numberOfItemWhenIncreased = 32;
+        // let numberOfItemWhenDecreased = 31;
+        fireEvent.click(screen.getByTestId("increase-items"));
+        await waitFor(() => userEvent.type(screen.getByTestId("increase-number-items"), "1"));
+        fireEvent.click(screen.getByText("Submit"));
+
+        await waitFor(() => {
+            expect(screen.getByTestId("boxview-number-items")).toBeInTheDocument();
+            expect(screen.getByTestId("boxview-number-items")).toHaveTextContent(
+                "# " + numberOfItemWhenIncreased,
+            );
+        });
+
+        // fireEvent.click(screen.getByTestId("decrease-items"));
```

Comment:
Do you plan to still implement this test?

Reply:
@pylipp it make mocking date quiet complex so I skip this for time being